src/Storages/Hive/StorageHive.cpp

```diff
@@ -115,19 +116,31 @@ class StorageHiveSource : public SourceWithProgress, WithContext
         , format(std::move(format_))
         , compression_method(compression_method_)
         , max_block_size(max_block_size_)
-        , sample_block(std::move(sample_block_))
+        , header_block(header_block_)
+        , sample_block(sample_block_)
```

Comment:
I think one sample_block should be enough

Reply:
`header_block` is used in the case when all columns in `sample_block` are partition columns, to find a actual column to read. Since there is no a easy way to read the num of row from the `ReadBuffer`.

Maybe there is a better implementation. I take a try.

Reply:
updated